doc/error-catalog.md

```diff
@@ -174,6 +174,35 @@ Examples:
       echo yes
     }
 
+### OILS-ERR-15
+
+Incorrect:
+
+    # Expression mode
+    if (a || b) {
+      echo yes
+    }
+
+    # Command mode
+    if test --dir a or test --dir b { ... }
```

Comment:
I would use the same style of 

```
if test --dir a || test --dir b { 
  echo yes
}

```

for the two examples ... just to make it consistent, and because `...` isn't literal YSH